specs/consensus/quint/statemachineAsync.qnt

```diff
@@ -224,13 +228,21 @@ action deliverSomeCertificate(v: Address) : bool = all {
 }
 
 
-// deliver a message or a certiciate. 
+// deliver a message or a certificate. 
 // Take it from the network buffer of from the faulty set and put it into the incoming sets
 action deliver(v: Address) : bool = any {
-    nondet prop = oneOf(propBuffer.get(v).union(AllFaultyProposals))
-    deliverProposal(v, prop),
-    nondet vote = oneOf(voteBuffer.get(v).union(AllFaultyVotes))
-    deliverVote(v, vote),
+    val props = propBuffer.get(v).union(AllFaultyProposals)
+    all{
```

Comment:
There is a `deliverSomeProposal` action in this file, with the same issue: it does not test if there is a proposal buffer to deliver.

Reply:
This method, however, does not delivery faulty proposals.

Reply:
The method is used in:

- ./tests/decideForPastRound/decideForPastRoundrun.qnt
- tests/line42/line42run.qnt